lgc/builder/BuilderBase.cpp

```diff
@@ -30,11 +30,25 @@
  */
 #include "lgc/util/BuilderBase.h"
 #include "llvm/IR/InlineAsm.h"
+#include "llvm/IR/IntrinsicInst.h"
 #include "llvm/IR/IntrinsicsAMDGPU.h"
 
 using namespace lgc;
 using namespace llvm;
 
+// =====================================================================================================================
+// Set the insert point to be just past the initial block of allocas in the given function's entry block.
+//
+// Use this method if you need to insert code to define values that are accessible in the entire function.
+void BuilderCommon::setInsertPointPastAllocas(Function &fn) {
+  BasicBlock &bb = fn.getEntryBlock();
+  auto it = bb.begin();
+  while (it != bb.end() &&
```

Comment:
nit: since this doesn't modify the end iterator, consider querying `bb.end()` once before the loop